poh/src/poh_recorder.rs

```diff
@@ -791,12 +795,20 @@ impl PohRecorder {
                 continue;
             };
 
-            // If the leader for this slot is not me, then it's the previous
-            // leader's last slot.
-            if leader_for_slot != *my_pubkey {
-                // Check if the last slot PoH reset onto was the previous leader's last slot.
-                return slot == self.start_slot();
+            if leader_for_slot == *my_pubkey {
+                // This is me... keep going.
+                continue;
             }
+
+            if !seen_previous_leader_slot {
+                // First slot we've seen from prior leader. Keep going.
+                seen_previous_leader_slot = true;
+                continue;
+            }
+
+            // We've identified the previous leader's second to last slot. Now
+            // see if this is what we're building off of.
+            return slot != self.start_slot();
```

Comment:
So if our reset bank was the leader's second-to-last block, we *cannot* skip grace ticks because we assume the leader is now working on their last block. Seems reasonable to me.

But it's technically possible (although probably very unlikely) that the previous leader is building their last block which descends from a block *other than* their second-to-last leader slot. In that case, we would determine that we *could* skip grace ticks but it's possible that the leader is working on their last block and we should have given grace ticks.

Reply:
yep, that's true